xonsh/completers/completer.py

```diff
@@ -125,3 +190,41 @@ def remove_completer(name: str):
         return
     else:
         return None, err + "\n", 1
+
+
+class CompleterAlias(ArgParserAlias):
+    """CLI to add/remove/list xonsh auto-complete functions"""
+
+    def build(self):
+        parser = self.create_parser(prog="completer")
+        parser.add_command(_register_completer, prog="add")
+        parser.add_command(remove_completer, prog="remove", aliases=["rm"])
+        parser.add_command(list_completers, prog="list", aliases=["ls"])
+        return parser
+
+
+completer_alias = CompleterAlias()
+
+
+@contextual_command_completer
+def complete_argparser_aliases(command: CommandContext):
+    """
+    Completion for "completer"
+    """
+    if not command.args:
+        return
+    cmd = command.args[0].value
+
+    alias = XSH.aliases.get(cmd)  # type: ignore
+    if not hasattr(alias, "parser"):
+        return
+
+    if command.suffix:
+        # completing in a middle of a word
+        # (e.g. "completer some<TAB>thing")
+        return
+
+    possible = complete_argparser(alias.parser, command=command, alias=alias)
+    for comp in possible:
+        if comp.startswith(command.prefix):
+            yield comp
```

Comment:
Yes that will be easier to implement. This version still has a todo left :)
I want to see all the options available for a command in one glance. that way I can decide on arguments that I want to use.

Reply:
The standard (in xonsh and other shells) is to only show these once the user has wrote the first dash, it's what users expect and I don't see any reason to do otherwise

Reply:
added an environment variable to control this behaviour

https://github.com/xonsh/xonsh/pull/4391/commits/db5a4f88bb624e1694429e8c80fae3ac5324afe6